src/tools/network.ts

```diff
@@ -44,11 +45,37 @@ const requests = defineTool({
   },
 });
 
-function renderRequest(request: playwright.Request, response: playwright.Response | null) {
+// Only include request/response body for XHR/fetch/json requests
+function shouldIncludeBody(request: playwright.Request): boolean {
+  const resourceType = request.resourceType();
+  if (resourceType === 'xhr' || resourceType === 'fetch')
+    return true;
+  const headers = request.headers();
+  if (headers['x-requested-with']?.toLowerCase() === 'xmlhttprequest')
+    return true;
+  return request.isNavigationRequest() === false &&
+         (headers.accept?.includes('application/json') ||
+          headers['content-type']?.includes('application/json'));
```

Comment:
let's just check if there's a body, not what its content type is. if the request has a body, that's always interesting.

Reply:
I fixed the conditions like the below

https://github.com/microsoft/playwright-mcp/pull/251/commits/996df4f125f3aa673563b6401588bc2490a306ba